server/plugins/surprise/plugin_surprise.cpp

```diff
@@ -28,6 +28,16 @@ void PluginSurprise::createCron(Bunny * b)
 	Cron::RegisterOneShot(this, GetRandomizedFrequency(frequency), b, QVariant(), NULL);
 }
 
+bool PluginSurprise::OnClick(Bunny * b, PluginInterface::ClickType type)
+{
```

Comment:
You should add a flag here

Else you can't have Randomized Surprises and Taichi::OnClick on the same time (Surprise::OnClick could be fired before Taichi)